Program.cs

```diff
@@ -1,2 +1,58 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿using DelegateHW;
+using System;
+
+namespace MyApp // Note: actual namespace depends on the project name.
+{
+    internal class Program
+    {
+        public static IEnumerable<T> Sort<T, K>(IEnumerable<T> source, Func<T, K> keySelector)
+        {
+            var items = source.ToArray();
+            var keys = items.Select(keySelector).ToArray();
```

Comment:
Why you use key selector instead of predicate?
Your signature should be Func<T, T, bool> comparer for sorting.